python/lsst/afw/geom/spanSet.cc

```diff
@@ -0,0 +1,225 @@
+                           &SpanSet::setMask<pixel>);
+    }
+
+    template <typename pixel, typename PyClass>
+    void declareClearMaskMethod(PyClass & cls) {
+        cls.def("clearMask",
+                (void (SpanSet::*)(lsst::afw::image::Mask<pixel> &, pixel) const) &SpanSet::clearMask<pixel>);
+    }
+
+    template <typename pixel, typename PyClass>
+    void declareIntersectMethod(PyClass & cls) {
+        cls.def("intersect", (std::shared_ptr<SpanSet> (SpanSet::*)(lsst::afw::image::Mask<pixel> const &,
+                                                                    pixel const &) const)
+                                                                    &SpanSet::intersect<pixel>);
+    }
+
+    template <typename pixel, typename PyClass>
+    void declareIntersectNotMethod(PyClass & cls) {
+        cls.def("intersectNot", (std::shared_ptr<SpanSet> (SpanSet::*)(lsst::afw::image::Mask<pixel> const &,
+                                                                       pixel const &) const)
+                                                                       &SpanSet::intersectNot<pixel>);
+    }
+
+    template <typename pixel, typename PyClass>
+    void declareUnionMethod(PyClass & cls) {
+        cls.def("union", (std::shared_ptr<SpanSet> (SpanSet::*)(lsst::afw::image::Mask<pixel> const &,
+                                                                pixel const &) const)
+                                                                &SpanSet::union_<pixel>);
+    }
+
+    template <typename pixel>
+    void declareMaskToSpanSetFunction(py::module & mod) {
+        mod.def("maskToSpanSet",
+                []
+                (lsst::afw::image::Mask<pixel> mask)
+                {
+                    return maskToSpanSet(mask);
+                });
+        mod.def("maskToSpanSet",
+                []
+                (lsst::afw::image::Mask<pixel> mask, pixel const & bitmask)
+                {
+                    auto functor = [&bitmask](pixel const & pixval){ return (pixval & bitmask) == bitmask; };
+                    return maskToSpanSet(mask, functor);
+                });
+    }
+
+    template <typename pixel, typename PyClass>
+    void declareMaskMethods(PyClass & cls) {
+        declareSetMaskMethod<pixel>(cls);
+        declareClearMaskMethod<pixel>(cls);
+        declareIntersectMethod<pixel>(cls);
+        declareIntersectNotMethod<pixel>(cls);
+        declareUnionMethod<pixel>(cls);
+    }
+
+} // end anonymous namespace
+
+PYBIND11_PLUGIN(_spanSet) {
+    using MaskPixel = lsst::afw::image::MaskPixel;
+    py::module mod("_spanSet", "Python wrapper for afw _spanSet library");
+
+    if(_import_array() < 0) {
+        PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+        return nullptr;
+    }
+
+    py::enum_<Stencil>(mod, "Stencil")
+        .value("CIRCLE", Stencil::CIRCLE)
+        .value("BOX", Stencil::BOX)
+        .value("MANHATTAN", Stencil::MANHATTAN);
```

Comment:
This works becuase the enum is supposed to be explicitly accessed as Stencil.(shape), and not be available at a higher namespace scope.